contracts/services/QueryService.sol

```diff
@@ -1,47 +1,237 @@
  // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "../managers/crowdfunding-softcap/ICfManagerSoftcapFactory.sol";
-import "../managers/crowdfunding-softcap/ICfManagerSoftcap.sol";
 import "../shared/Structs.sol";
+import "../shared/ICampaignFactoryCommon.sol";
+import "../shared/IAssetFactoryCommon.sol";
+import "../shared/ICampaignCommon.sol";
+import "../shared/IAssetCommon.sol";
+import "../registry/INameRegistry.sol";
 
 contract QueryService {
 
-    struct CampaignWithInvestmentState {
-        Structs.CfManagerSoftcapState campaign;
-        uint256 tokenAmount;
-        uint256 tokenValue;
+    function getCampaignsForIssuerName(
+        string memory issuerName,
+        address[] memory factories,
+        INameRegistry nameRegistry
+    ) public view returns (Structs.CampaignCommonStateWithName[] memory) {
+        address issuer = nameRegistry.getIssuer(issuerName);
+        return getCampaignsForIssuer(issuer, factories, nameRegistry);
     }
-
+    
     function getCampaignsForIssuer(
         address issuer,
-        address cfManagerFactoryAddress
-    ) external view returns (Structs.CfManagerSoftcapState[] memory) {
-        ICfManagerSoftcapFactory cfManagerFactory = ICfManagerSoftcapFactory(cfManagerFactoryAddress);
-        address[] memory instances = cfManagerFactory.getInstancesForIssuer(issuer);
-        Structs.CfManagerSoftcapState[] memory mapped = new Structs.CfManagerSoftcapState[](instances.length);
-        for (uint256 i = 0; i < instances.length; i++) { mapped[i] = ICfManagerSoftcap(instances[i]).getState(); }
-        return mapped;
+        address[] memory factories,
+        INameRegistry nameRegistry
+    ) public view returns (Structs.CampaignCommonStateWithName[] memory) {
+        if (factories.length == 0) { return new Structs.CampaignCommonStateWithName[](0); }
+        
+        uint256 totalItems = 0;
+        uint256[] memory instanceCountPerFactory = new uint256[](factories.length);
```

Comment:
Do we really need `uint256`? Looks to me that `uint8` would be good enough for everything.